packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -4754,3 +4757,57 @@ export class Gitxnas extends Gtxnas {
 		super.execute(stack);
 	}
 }
+
+/**
+ * Takes the last value from stack and if base64encoded, decodes it acording to the
+ * encoding e and pushes it back to the stack, otherwise throws an error
+ */
+export class Base64Decode extends Op {
+	readonly line: number;
+	readonly encoding: Base64Encoding;
+
+	/**
+	 * Asserts 1 argument is passed.
+	 * @param args Expected arguments: [e], where e = {URLEncoding, StdEncoding}.
+	 * @param line line number in TEAL file
+	 */
+	constructor(args: string[], line: number) {
+		super();
+		this.line = line;
+		assertLen(args.length, 1, line);
+		const argument = args[0];
+		switch (argument) {
+			case "URLEncoding": {
+				this.encoding = 0;
+				break;
+			}
+			case "StdEncoding": {
+				this.encoding = 1;
+				break;
+			}
+			default: {
+				throw new RuntimeError(RUNTIME_ERRORS.TEAL.UNKNOWN_ENCODING, {
+					encoding: argument,
+					line: this.line,
+				});
+			}
+		}
+	}
+
+	execute(stack: TEALStack): void {
+		this.assertMinStackLen(stack, 1, this.line);
+		const last = this.assertBytes(stack.pop(), this.line);
+		const enc = new TextDecoder("utf-8");
+		const decoded = enc.decode(last);
+		switch (this.encoding) {
+			case Base64Encoding.URL:
+				assertBase64Url(convertToString(last), this.line);
+				stack.push(new Uint8Array(Buffer.from(decoded.toString(), "base64url")));
```

Comment:
Why we don't store "base64Url" and "base64" to variable? 
We can store it to `this.encoding`. 
Another questions is when I check Opcode class I also see few assert function in Opcode class. 
Should we move `asserBase64URL` and `assertBase64` to Opcode class?